Source/WebKit/UIProcess/API/glib/WebKitWebExtension.cpp

```diff
@@ -0,0 +1,1156 @@
+{
+    HashMap<String, Ref<API::Data>> newResources;
+    for (const auto& it : resources)
+        newResources.add(it.key, API::Data::createWithoutCopying(it.value));
+
+    Ref extension = WebKit::WebExtension::create(WTFMove(newResources));
+
+    if (!extension->errors().isEmpty()) {
+        Ref internalError = extension->errors().last();
+        g_set_error(error, webkit_web_extension_error_quark(),
+            toWebKitWebExtensionError(internalError->errorCode()), internalError->localizedDescription().utf8().data(), nullptr);
+    }
+
+    WebKitWebExtension* object = WEBKIT_WEB_EXTENSION(g_object_new(WEBKIT_TYPE_WEB_EXTENSION, nullptr));
+    object->priv->extension = WTFMove(extension);
+    return object;
+}
+
+/**
+ * webkit_web_extension_get_manifest_version:
+ * @extension: a #WebKitWebExtension
+ *
+ * Get the parsed manifest version, or `0` if there is no
+ * version specified in the manifest.
+ *
+ * An ``WKWebExtensionErrorUnsupportedManifestVersion`` error will be
+ * reported if the manifest version isn't specified.
+ * 
+ * Returns: the parsed manifest version.
+ * 
+ * Since: 2.48
+ */
+gdouble webkit_web_extension_get_manifest_version(WebKitWebExtension* extension)
+{
+    g_return_val_if_fail(WEBKIT_IS_WEB_EXTENSION(extension), 0);
+
+    return extension->priv->extension->manifestVersion();
+}
+
+/**
+ * webkit_web_extension_supports_manifest_version:
+ * @extension: a #WebKitWebExtension
+ * @manifest_version: the version number to check
+ * 
+ * Checks if a manifest version is supported by the extension.
+ * 
+ * Returns: `TRUE` if the extension specified a manifest version
+ * that is greater than or equal to @manifest_version.
+ * 
+ * Since: 2.48
+ */
+gboolean webkit_web_extension_supports_manifest_version(WebKitWebExtension* extension, gdouble manifestVersion)
+{
+    g_return_val_if_fail(WEBKIT_IS_WEB_EXTENSION(extension), FALSE);
+
+    return extension->priv->extension->supportsManifestVersion(manifestVersion);
+}
+
+/**
+ * webkit_web_extension_get_default_locale:
+ * @extension: a #WebKitWebExtension
+ *
+ * Get the default locale for the extension.
+ * 
+ * Returns: the default locale, or %NULL if there was no default
+ * locale specified.
+ * 
+ * Since: 2.48
+ */
+const gchar* webkit_web_extension_get_default_locale(WebKitWebExtension* extension)
+{
+    g_return_val_if_fail(WEBKIT_IS_WEB_EXTENSION(extension), nullptr);
+
+    WebKitWebExtensionPrivate* priv = extension->priv;
+    if (!priv->defaultLocale.isNull())
+        return priv->defaultLocale.data();
+
+    auto defaultLocale = priv->extension->defaultLocale();
+    if (defaultLocale.isEmpty())
+        return nullptr;
+
+    priv->defaultLocale = defaultLocale.utf8();
+    return priv->defaultLocale.data();
+}
+
+/**
+ * webkit_web_extension_get_display_name:
+ * @extension: a #WebKitWebExtension
+ *
+ * Get the localized name for the extension.
+ * 
+ * Returns: the localized name, or %NULL if there was no
```

Comment:
Same here, this needs a `(nullable)` annotation. By default only `gpointer` and closure data parameters are considered nullable ([docs](https://gi.readthedocs.io/en/latest/annotations/giannotations.html#nullable-parameters)). Please do check the rest of functions and add the annotation where needed.